libs/checkpoint/docs/index.rst

```diff
@@ -0,0 +1,136 @@
+``checkpoint``. All other policies passed to ``save_checkpoint`` will return a
+``future`` to a ``checkpoint``.
+
+Sometimes ``checkpoint`` s must be declared before they are used.
+``save_checkpoint`` allows users to move pre-created ``checkpoint`` s into the
+function as long as they are the first container passing into the function (In
+the case where a launch policy is used, the ``checkpoint`` will immediately
+follow the launch policy). An example of these features can be found below:
+
+.. literalinclude:: ../tests/unit/checkpoint.cpp
+   :language: c++
+   :lines: 27-38
+
+Now that we can create ``checkpoint`` s we now must be able to restore the
+objects they contain into memory. This is accomplished by the function
+``restore_checkpoint``. This function takes a ``checkpoint`` and fills its data
+into the containers it is provided. It is important to remember that the
+containers must be ordered in the same way they were placed into the
+``checkpoint``. For clarity see the example below:
+
+.. literalinclude:: ../tests/unit/checkpoint.cpp
+   :language: c++
+   :lines: 41-49
+
+The core utility of ``checkpoint`` is in its ability to make certain data
+persistent. Often this means that the data is needed to be stored in an object,
+such as a file, for later use. For these cases we have provided two solutions:
+stream operator overloads and access iterators.
+
+We have created the two stream overloads
+``operator<<`` and ``operator>>`` to stream data
+out of and into ``checkpoint``. You can see an
+example of the overloads in use below:
+
+.. literalinclude:: ../tests/unit/checkpoint.cpp
+   :language: c++
+   :lines: 176-186
+
+This is the primary way to move data into and out of a ``checkpoint``. It is
+important to note, however, that users should be cautious when using a stream
+operator to load data an another function to remove it (or vice versa). Both
+``operator<<`` and ``operator>>`` rely on a ``.write()`` and a ``.read()``
+function respectively. In order to know how much data to read from the
+``std::istream``, the ``operator<<`` will write the size of the ``checkpoint``
+before writing the ``checkpoint`` data. Correspondingly, the ``operator>>`` will
+read the size of the stored data before reading the data into new instance of
+``checkpoint``. As long as the user employs the ``operator<<`` and
+``operator>>`` to stream the data this detail can be ignored.
+
+.. important::
+
+   Be careful when mixing ``operator<<`` and ``operator>>`` with other
+   facilities to read and write to a ``checkpoint``. ``operator<<`` writes and
+   extra variable and ``operator>>`` reads this variable back separately. Used
+   together the user will not encounter any issues and can safely ignore this
+   detail.
+
+Users may also move the data into and out of a ``checkpoint`` using the exposed
+``.begin()`` and ``.end()`` iterators. An example of this use case is
+illustrated below.
+
+.. literalinclude:: ../tests/unit/checkpoint.cpp
```

Comment:
This needs to be
```suggestion
.. literalinclude:: ../../../../libs/checkpoint/tests/unit/checkpoint.cpp
```
and a link needs to be added to the libs directory: https://github.com/STEllAR-GROUP/hpx/blob/f5b348e053be7433e215f996d6ec41e428047938/docs/CMakeLists.txt#L258-L265.

Reply:
@msimberg I have applied those changes, could you please check again?